src/operations/client_bulk_write/client_bulk_write.ts

```diff
@@ -37,9 +37,34 @@ export class ClientBulkWriteOperation extends CommandOperation<ClientBulkWriteCu
     server: Server,
     session: ClientSession | undefined
   ): Promise<ClientBulkWriteCursorResponse> {
-    return await super.executeCommand(server, session, this.command, ClientBulkWriteCursorResponse);
+    let command;
+
+    if (server.description.type === ServerType.LoadBalancer) {
+      // Checkout a connection to build the command.
+      const connection = await server.pool.checkOut();
+      command = this.commandBuilder.buildBatch(
+        connection.hello?.maxMessageSizeBytes,
+        connection.hello?.maxWriteBatchSize
+      );
+      // Pin the connection to the session so it get used to execute the command and we do not
+      // perform a double check-in/check-out.
+      session?.pin(connection);
```

Comment:
```typescript
const connection = await server.pool.checkOut();
session?.pin(connection);
command = this.commandBuilder.buildBatch(
  connection.hello?.maxMessageSizeBytes,
  connection.hello?.maxWriteBatchSize
);
```

flip the order, perhaps, in case buildBatch throws for some reason?

Also - I don't think session can ever be undefined but if it is, we'd leak a connection.  Should we maybe assert that it's defined?

Reply:
I've updated per the suggestions here.